### PR TITLE
Improve display of run details on the status page

### PR DESCRIPTION
**Pull request by jonavellecuerdo**
### Purpose and background context
Improve display of "Run details" on task status page. 

### How can a reviewer manually see the effects of these changes?

**IMPORTANT**: Review via screenshots is sufficient! Now that the command args are updated and since Dev1 and Stage are connected to Alma Sandbox and emails are sent for both "review" and "final" runs, will need to minimize the number of test runs for the time being 😅 . Will be putting in a ticket for future work to avoid sending emails when testing!

Run details **before the change**: 
Screenshot 2024-09-19 at 8 38 39 AM

Run details **after the change**: 

image

**If no invoices**

image

### Includes new or updated dependencies?
NO

### Changes expectations for external applications?
NO

### What are the relevant tickets?
https://mitlibraries.atlassian.net/browse/IN-1066

### Developer
- [ ] All new ENV is documented in README
- [ ] All new ENV has been added to staging and production environments
- [ ] All related Jira tickets are linked in commit message(s)
- [ ] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed and verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: webapp/utils/aws/cloudwatch.py [hunk 1]
```diff
@@ -39,9 +39,21 @@ def client(self) -> "CloudWatchLogsClientType":
     def get_log_messages(self, task_id: str) -> list:
         messages: list = []
         if logs := self.get_log_events(task_id):
-            messages.extend(event["message"] for event in logs)
+            return self.get_log_summary(logs)
         return messages
 
+    def get_log_summary(self, logs: list[dict]) -> list[str]:
```
Comment (ghukill) [inline webapp/utils/aws/cloudwatch.py:45]: I probably sound like a broken record, but I think a little docstring could be helpful here.

This is a super elegant update (nice work!), but I'll admit it took me a bit to parse the function and understand how it worked (that given a target log line, we would _then_ return all lines after that via `logs[summer_index:]]`).

But again, seems like a great solution, and improved run details!  Just think a little blurb here explaining could be helpful.

**Review by ghukill — APPROVED** [review]
Enthusiastic approve.  Nice work.

Left a comment that a docstring for `get_log_summary()` could be nice, but I'm comfortable approving either way.

File: webapp/utils/aws/cloudwatch.py (hunk 1 above)
Comment (ehanson8) [reply webapp/utils/aws/cloudwatch.py:45]: Agreed!

**Review by ehanson8 — APPROVED** [review]
Agree on the potential value of a docstring but also approved!

File: webapp/utils/aws/cloudwatch.py [hunk 2]
```diff
@@ -43,6 +43,14 @@ def get_log_messages(self, task_id: str) -> list:
         return messages
 
     def get_log_summary(self, logs: list[dict]) -> list[str]:
+        """Get summary of SAP invoice processing logs.
+
+        This function will first determine the index of the log event
+        the marks the start of the "summary" log messages that
+        describe the output of the SAP invoice processing run.
+        The function will then retrieve all the messages starting from
+        that index, effectively retrieving a summary of the run.
```
Comment (ghukill) [inline webapp/utils/aws/cloudwatch.py:52]: Excellent addition - thanks!

**Review by ghukill — APPROVED** [review]

File: webapp/utils/aws/cloudwatch.py (hunk 2 above)
Comment (ehanson8) [reply webapp/utils/aws/cloudwatch.py:52]: Agreed!

**Review by ehanson8 — APPROVED** [review]
Approval stands!

**Pull request merged**